packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -493,22 +506,124 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
+  final GlobalKey fieldKey;
+
+  /// The size of the field in [RawAutocomplete.fieldViewBuilder].
+  final Size fieldSize;
+
+   /// A direction in which to open the options view overlay.
+  final OptionsViewOpenDirection optionsViewOpenDirection;
+
+  /// The [BuildContext] of the [Overlay] that options are rendered in.
+  final BuildContext overlayContext;
+
+  /// The [TextDirection] of this part of the widget tree.
+  final TextDirection textDirection;
+
+  // Eyeballed to be big enough for about one item in the default
+  // Autocomplete.optionsViewBuilder. The assumption is that the user likely
+  // wants the list of options to move to stay on the screen rather than get any
+  // smaller than this. Allows Autocomplete to work when it has very little
+  // screen height available (as in b/317115348) by positioning itself on top of
+  // the field, while in other cases to size itself based on the height under
+  // the field.
+  static const double _kMinUsableHeight = 52.0;
+
+  Offset get _fieldOffset {
+    final RenderBox? fieldRenderBox = fieldKey.currentContext?.findRenderObject() as RenderBox?;
+    if (fieldRenderBox == null) {
+      return Offset.zero;
+    }
+    return fieldRenderBox.localToGlobal(
+      Offset.zero,
+      ancestor: Overlay.of(overlayContext).context.findRenderObject(),
+    );
+  }
+
+  // Limits the child to the space above/below the field, with a minimum, and
+  // with the same maxWidth constraint as the field has.
+  @override
+  BoxConstraints getConstraintsForChild(BoxConstraints constraints) {
+    final Offset fieldOffset = _fieldOffset;
+    return constraints.loosen().copyWith(
+      maxWidth: fieldBoxConstraints.maxWidth,
+      maxHeight: switch (optionsViewOpenDirection) {
+        OptionsViewOpenDirection.down => max(
+          min(_kMinUsableHeight, constraints.maxHeight),
+          constraints.maxHeight - fieldOffset.dy - fieldSize.height,
         ),
+        OptionsViewOpenDirection.up => max(_kMinUsableHeight, fieldOffset.dy),
+      },
+    );
+  }
+
+  // Positions the child above/below the field and aligned with the left/right
+  // side based on text direction.
+  @override
+  Offset getPositionForChild(Size size, Size childSize) {
+    final Offset fieldOffset = _fieldOffset;
+    return Offset(
+      switch (textDirection) {
+        TextDirection.ltr => fieldOffset.dx,
+        TextDirection.rtl => fieldOffset.dx + fieldSize.width - childSize.width,
+      },
+      clampDouble(
+        switch (optionsViewOpenDirection) {
+          OptionsViewOpenDirection.down => fieldOffset.dy + fieldSize.height,
+          OptionsViewOpenDirection.up => fieldOffset.dy - childSize.height,
+        },
+        0.0,
+        max(0.0, size.height - childSize.height),
       ),
     );
   }
+
+  @override
+  bool shouldRelayout(_OptionsLayoutDelegate oldDelegate) {
+    // This class depends on the position of the field (_fieldOffset) in order
```

Comment:
That sounds like you should use `CompositeTransformFollower`?

Reply:
See https://github.com/flutter/flutter/pull/143249#discussion_r1649449912